### PR TITLE
Create login page

### LINKED ISSUE #92: Create login page

### PR DESCRIPTION
**Pull request by Mojtabarezaei4**
Maybe we can have a header instead of `LogoComponent` which might be for all the pages?

close #92

### CODE REVIEW

File: Queueomatic/Client/Components/Login/LoginComponent.razor [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <Authorized>
+        <RedirectHome/>
+    </Authorized>
+</AuthorizeView>
+<div class="component">
+    <div class="form-container">
+        <p class="title">Login</p>
+        <EditForm class="form" Model="_loginDto" OnValidSubmit="Login">
+            <div class="input-group">
+                <label for="username">Email</label>
```
Comment (FabianAronsson) [inline Queueomatic/Client/Components/Login/LoginComponent.razor:17]: This label is incorrect. It should be pointing to the id of the input field

File: Queueomatic/Client/Components/Login/LoginComponent.razor [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                <input type="email" name="username" placeholder="" id="input-field" @bind-value="_loginDto.Email" required>
```
Comment (FabianAronsson) [inline Queueomatic/Client/Components/Login/LoginComponent.razor:18]: Change the id to a class and add an id that matches the for in the label above

File: Queueomatic/Client/Components/Login/LoginComponent.razor [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                <div class="invalid-feedback">
+                    Please provide your email.
+                </div>
+            </div>
+            <div class="input-group">
+                <label for="password">Password</label>
+                <input type="password" name="password" placeholder="" id="input-field" @bind-value="_loginDto.Password" required>
```
Comment (FabianAronsson) [inline Queueomatic/Client/Components/Login/LoginComponent.razor:25]: Same as above

File: Queueomatic/Client/Components/Login/LoginComponent.razor [hunk 4]
```diff
@@ -0,0 +1,46 @@
+﻿@using Blazored.SessionStorage
+@inject HttpClient _httpClient;
+@inject NavigationManager _navigationManager;
```
Comment (FabianAronsson) [inline Queueomatic/Client/Components/Login/LoginComponent.razor:3]: These fields are not private. Change their naming convention

File: Queueomatic/Client/Components/Login/LoginComponent.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Net.Http.Json;
+using System.Text.Json;
+using Microsoft.AspNetCore.Components;
+using Queueomatic.Shared.DTOs;
+
+namespace Queueomatic.Client.Components.Login;
+
+public partial class LoginComponent : ComponentBase
```
Comment (FabianAronsson) [inline Queueomatic/Client/Components/Login/LoginComponent.cs:8]: This cs file should use code isolation like the css file uses.

File: Queueomatic/Client/Components/Login/LoginComponent.razor.css [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    font-size: 2rem;
+    line-height: 2rem;
+    font-weight: 700;
+}
+
+.form {
+    margin-top: 1.5rem;
+}
+
+#input-field {
```
Comment (FabianAronsson) [inline Queueomatic/Client/Components/Login/LoginComponent.razor.css:43]: Change to class

**Review by FabianAronsson — CHANGES_REQUESTED** [review]
All in all, looks great! Perhaps a fade in animation on component load for the login view would be nice?

File: Queueomatic/Client/Components/Login/LoginComponent.cs (hunk 5 above)
Comment (Mojtabarezaei4) [reply Queueomatic/Client/Components/Login/LoginComponent.cs:8]: What do you mean by this?

File: Queueomatic/Client/Components/Login/LoginComponent.razor (hunk 4 above)
Comment (Mojtabarezaei4) [reply Queueomatic/Client/Components/Login/LoginComponent.razor:3]: What naming convention are you suggesting? 

I think this is more readable. These work as fields and it's easier to read and understand.

File: Queueomatic/Client/Components/Login/LoginComponent.cs (hunk 5 above)
Comment (Mojtabarezaei4) [reply Queueomatic/Client/Components/Login/LoginComponent.cs:8]: Oh, I see now. It is wrong, I will fix it.

File: Queueomatic/Client/Components/Login/LoginComponent.razor (hunk 2 above)
Comment (Mojtabarezaei4) [reply Queueomatic/Client/Components/Login/LoginComponent.razor:18]: Should it? We don't have an id. We use userName which is userEmail as id.

File: Queueomatic/Client/Components/Login/LoginComponent.razor (hunk 1 above)
Comment (Mojtabarezaei4) [reply Queueomatic/Client/Components/Login/LoginComponent.razor:17]: Should it? We don't have an id. We use userName which is userEmail as id.

Comment (FabianAronsson) [reply Queueomatic/Client/Components/Login/LoginComponent.razor:17]: Yes it should. This is how forms work in HTML

File: Queueomatic/Client/Components/Login/LoginComponent.razor (hunk 4 above)
Comment (FabianAronsson) [reply Queueomatic/Client/Components/Login/LoginComponent.razor:3]: They are properties, and should be treated as such. The are usually capitalized on their first letter.

**Review by FabianAronsson — APPROVED** [review]

**Pull request merged**